runtime/drivers/sqlite/registry.go

```diff
@@ -91,3 +117,13 @@ func (c *connection) DeleteInstance(_ context.Context, id string) error {
 	_, err := c.db.ExecContext(ctx, "DELETE FROM instances WHERE id=$1", id)
 	return err
 }
+
+func toJSON(data map[string]string) ([]byte, error) {
+	return json.Marshal(data)
+}
+
+func fromJSON(data []byte) (map[string]string, error) {
+	m := make(map[string]string)
+	err := json.Unmarshal(data, &m)
+	return m, err
```

Comment:
Change to `var m map[string]string`. The `json.Unmarshal` call will alloc the map if needed (checked [here](https://go.dev/play/p/_lkVWfStaNW)), and this will ensure that if there's an `err`, `m` is `nil` (it's not good to return both an error and non-nil value)